Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/Microsoft_Defender_For_Endpoint_-_Collect_investigation_package_.yml

```diff
@@ -0,0 +1,440 @@
+      script: '|||microsoft-atp-get-investigation-package-sas-uri'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "10"
+    scriptarguments:
+      action_id:
+        simple: ${MicrosoftATP.MachineAction.ID}
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 390,
+          "y": 1230
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "6":
+    id: "6"
+    taskid: 6a3b3286-33d2-4623-80d2-3af90fae2c8f
+    type: regular
+    task:
+      id: 6a3b3286-33d2-4623-80d2-3af90fae2c8f
+      version: -1
+      name: Validate endpoint input
+      description: Returns information about an endpoint.
+      script: '|||endpoint'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "7"
+    scriptarguments:
+      hostname:
+        simple: ${inputs.Hostnames}
+      id:
+        simple: ${inputs.MachineIDs}
+      ip:
+        simple: ${inputs.IPs}
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 162.5,
+          "y": 215
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "7":
+    id: "7"
+    taskid: 5e732682-dfe9-4fa0-8c75-a78f81104a92
+    type: condition
+    task:
+      id: 5e732682-dfe9-4fa0-8c75-a78f81104a92
+      version: -1
+      name: Are there any windows machine IDs?
+      type: condition
+      description: Are there any windows machine IDs?
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "8"
+      "yes":
+      - "2"
+    separatecontext: false
+    conditions:
+    - label: "yes"
+      condition:
+      - - operator: isNotEmpty
+          left:
+            value:
+              simple: Endpoint.ID
+            iscontext: true
+          right:
+            value: {}
+    view: |-
+      {
+        "position": {
+          "x": 162.5,
+          "y": 370
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "8":
+    id: "8"
+    taskid: 59afffe4-dd31-45ee-8111-af067040c394
+    type: title
+    task:
+      id: 59afffe4-dd31-45ee-8111-af067040c394
+      version: -1
+      name: Done
+      type: title
+      iscommand: false
+      brand: ""
+      description: ''
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 162.5,
+          "y": 1560
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "9":
```

Comment:
Dont we have a polling command?

Reply:
Not inside the command itself - the thing is for adding polling command you need to combine several commands (Request Pkg+ Check that the action was successfully ended + retrieve the file). 
So the polling here is just to validate the action status of the retrieve command.

Reply:
No problem to combine several API's to cover this
This was already done for other commands
Just need to define it for dev